editor/RubyScanner.cpp

```diff
@@ -140,7 +140,7 @@ bool Scanner::didBlockStart()
 
 bool Scanner::didBlockEnd()
 {
-    static const QRegExp regex(QLatin1String("24_"));
+    static const QRegExp regex(QLatin1String("(24|29)_"));
```

Comment:
The case you described already works.

Your patch break the indent of `def foo()`, what isn't covered yet are blocks starting with `{` and `}` and arrays.